megatron/megatron/core/pipeline_parallel/schedules.py

```diff
@@ -1823,14 +1825,27 @@ def enable_grad_sync():
             if num_warmup_microbatches == 0 and last_iteration:
                 if config.grad_sync_func is None or rank == 0:
                     enable_grad_sync()
+            
+            # For BWF pattern or in rank 0, we don't split W and B for reasons below.
+            #   1. to leverage batched p2p op (send_backward_recv_forward)
+            #   2. to overlap grad all-reduce for tensor parallel
+            #   3. to avoid redoing grad all-gather for sequence parallel
```

Comment:
Add a new schedule function in `flagscale` folder to avoid changing the Megtron files.